core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -154,18 +115,58 @@ public void close() throws IOException {
         IOUtils.close(closeables);
     }
 
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        IngestMetadata ingestMetadata = event.state().getMetaData().custom(IngestMetadata.TYPE);
+        if (ingestMetadata == null) {
+            return;
+        }
+
+        Map<String, Pipeline> pipelines = new HashMap<>();
+        for (Map.Entry<String, Map<String, Object>> entry : ingestMetadata.getPipelines().entrySet()) {
+            String id = entry.getKey();
+            try {
+                @SuppressWarnings("unchecked")
+                Map<String, Object> pipelineConfigCopy = (Map<String, Object>) deepCopy(entry.getValue());
+                pipelines.put(id, constructPipeline(id, pipelineConfigCopy));
```

Comment:
with BytesReference though what would we pass to the Pipeline.factory /ConfigurationUtils? Maybe have this PipelineConfiguration class, which also allows to retrieve its config in map format to hand it over to ConfigurationUtils?